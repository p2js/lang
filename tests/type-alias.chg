
type Point<T> = { x: T, y: T };
type Nested<T> = { p: Point<T> };

fn point<T>(x: T, y: T): Point<T> {
    return .{ x: x, y: y };
}

fn single<T>(v: T): Nested<T> {
    return .{ p: point(v, v) };
}

fn main(): i32 {
    // todo: actually use point and single here to test their codegen
    return 0;
}
