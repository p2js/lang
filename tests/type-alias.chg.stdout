expr @ 6:18 6:19 => T
expr @ 6:24 6:25 => T
expr @ 6:12 6:27 => {x: T,y: T}
expr @ 6:5 6:27 => never
expr @ 5:35 8:0 => ()
expr @ 10:18 10:23 => fn point(...)
expr @ 10:24 10:25 => T
expr @ 10:27 10:28 => T
expr @ 10:18 10:29 => Point<T>
expr @ 10:12 10:31 => {p: Point<T>}
expr @ 10:5 10:31 => never
expr @ 9:31 12:0 => ()
expr @ 15:12 15:13 => i32
expr @ 15:5 15:13 => never
expr @ 13:16 17:1 => ()



define i32 @main() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
store i32 0, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}


